pkg/controller/replication/replication_controller.go

```diff
@@ -280,14 +281,33 @@ func (r *ReconcileReplication) buildReplicationFromSpec(originalReplication *reg
 		return h.ReplicationPolicy{}, fmt.Errorf("both dest_registry and src_registry are set! Please specify only one of them")
 	}
 	if originalReplication.Spec.SrcRegistry != nil {
-		newRep.SrcRegistry = originalReplication.Spec.SrcRegistry.ToHarborRegistry()
+		hReg, err := r.getHarborRegistryFromRegistry(context.Background(), originalReplication.Spec.SrcRegistry, originalReplication.Namespace)
+		if err != nil {
+			return h.ReplicationPolicy{}, err
+		}
+		newRep.SrcRegistry = hReg
 	} else if originalReplication.Spec.DestRegistry != nil {
-		newRep.DestRegistry = originalReplication.Spec.DestRegistry.ToHarborRegistry()
+		hReg, err := r.getHarborRegistryFromRegistry(context.Background(), originalReplication.Spec.DestRegistry, originalReplication.Namespace)
+		if err != nil {
+			return h.ReplicationPolicy{}, err
+		}
+		newRep.DestRegistry = hReg
 	}
 
 	return newRep, nil
 }
 
+// getHarborRegistryFromRegistry retrieves the registryRef and returns a pointer to a goharbor-client Registry Object
+func (r *ReconcileReplication) getHarborRegistryFromRegistry (ctx context.Context, registryRef *v1.LocalObjectReference, namespace string) (*h.Registry, error) {
```

Comment:
Why `getHarborRegistryFromRegistry`? I think `getHarborRegistryFromRef` or similar would be a more conclusive name, as this method returns a `*h.Registry` based of a `LocalObjectReference`.
```suggestion
func (r *ReconcileReplication) getHarborRegistryFromRef (ctx context.Context, registryRef *v1.LocalObjectReference, namespace string) (*h.Registry, error) {
```